documentscanner/src/main/java/nz/mega/documentscanner/openCV/CropDetector.kt

```diff
@@ -0,0 +1,146 @@
+    }
+
+    private fun getPoints(src: Mat): List<MatOfPoint2f> {
+        // Blur the image to filter out the noise.
+        val blurred = Mat()
+        Imgproc.medianBlur(src, blurred, 9)
+
+        // Set up images to use.
+        val gray0 = Mat(blurred.size(), CvType.CV_8U)
+        val gray = Mat()
+
+        // For Core.mixChannels.
+        val contours: List<MatOfPoint> = ArrayList()
+        val rectangles: MutableList<MatOfPoint2f> = ArrayList()
+        val sources: MutableList<Mat> = ArrayList()
+        sources.add(blurred)
+        val destinations: MutableList<Mat> = ArrayList()
+        destinations.add(gray0)
+
+        // To filter rectangles by their areas.
+        val srcArea = src.rows() * src.cols()
+
+        // Find squares in every color plane of the image.
+        for (c in 0..2) {
+            val ch = intArrayOf(c, 0)
+            val fromTo = MatOfInt(*ch)
+            Core.mixChannels(sources, destinations, fromTo)
+
+            // Try several threshold levels.
+            for (l in 0 until THRESHOLD_LEVEL) {
+                if (l == 0) {
+                    // HACK: Use Canny instead of zero threshold level.
+                    // Canny helps to catch squares with gradient shading.
+                    // NOTE: No kernel size parameters on Java API.
+                    Imgproc.Canny(gray0, gray, 10.0, 20.0)
+
+                    // Dilate Canny output to remove potential holes between edge segments.
+                    Imgproc.dilate(gray, gray, Mat.ones(Size(3.0, 3.0), 0))
+                } else {
+                    val threshold = (l + 1) * 255 / THRESHOLD_LEVEL
+                    Imgproc.threshold(
+                        gray0,
+                        gray,
+                        threshold.toDouble(),
+                        255.0,
+                        Imgproc.THRESH_BINARY
+                    )
+                }
+
+                // Find contours and store them all as a list.
+                Imgproc.findContours(
+                    gray,
+                    contours,
+                    Mat(),
+                    Imgproc.RETR_LIST,
+                    Imgproc.CHAIN_APPROX_SIMPLE
+                )
+                for (contour in contours) {
+                    val contourFloat = OpenCvUtils.toMatOfPointFloat(
+                        contour
+                    )
+                    val arcLen = Imgproc.arcLength(contourFloat, true) * 0.02
+
+                    // Approximate polygonal curves.
+                    val approx = MatOfPoint2f()
+                    Imgproc.approxPolyDP(contourFloat, approx, arcLen, true)
+                    if (isRectangle(approx, srcArea)) {
+                        rectangles.add(approx)
+                    }
+                }
+            }
+        }
+        return rectangles
+    }
+
+    private fun isRectangle(polygon: MatOfPoint2f, srcArea: Int): Boolean {
```

Comment:
Add method documentation.

Reply:
Methods on this class are related with the OpenCV algorithm inherited from the forked lib. It was created by the original author who separated the logic based on his own criteria.

Reply:
Could we have at least a minimum documentation explaining that and the parameters meaning?